### DISCUSSION TITLE
CRITICAL: Negative values in Adj. Close data

### DISCUSSION THREAD

**Discussion by uzala92** [ranaroussi/yfinance#1165] [Development]
Dear developers,

I have detected an issue when i was getting the data of ticker "DANR.MI". It can be seen that in the beginning, besides having an adjusted close with disproportionate values, there are also negative values, which should never happen. I take the data using the following code:

```
stock = yf.Ticker("DANR.MI")
data = stock.history(period="max", interval="1d", auto_adjust=False, actions=True, rounding=False)
```

I think it may be a bug in yahoo finance data, anyway, do you have any idea how to avoid this problem?

Thanks in advance :)

**Comment by ValueRaider** [comment]
If it's critical then I recommend visiting Yahoo.com and inspecting their data.

**Reply by uzala92** [reply]
I did, yahoo data is wrong. That's why I was asking if anyone have any idea how to avoid this problem.

I mean, how to recalculate or do something with the wrong adjusted data. Maybe calculate it in another way or have a way to detect or solve the problem when appears.

Thanks!

**Reply by ValueRaider** [reply]
Don't use the adjusted price.  `auto_adjust=False`.

Maybe tell Yahoo.

**Comment by ValueRaider** [comment]
I've had another look at this and behaviour appears correct. The dividends before 2002 are much larger than price so adjusted price must be negative, and [investing.com](https://www.investing.com/equities/danieli---c-officine-meccaniche-dividends) confirms dividend size.

To do anything else requires evidence for dividend is wrong, or evidence "this should never happen".
